ui_framework/components/collapse_button/collapse_button.js

```diff
@@ -0,0 +1,29 @@
+import React, {
+  PropTypes,
+} from 'react';
+import classNames from 'classnames';
+
+const DIRECTIONS = [
+  'down',
+  'up',
+  'left',
+  'right'
+];
+
+export const KuiCollapseButton = ({ className, direction, ...rest }) => {
+  const classes = classNames('kuiCollapseButton', className);
+  const childClasses = classNames('kuiIcon', `fa-chevron-circle-${direction}`);
+
+  return (<button
+            type="button"
+            className={classes}
+            {...rest}
+          >
+            <div className={childClasses}/>
+          </button>);
+};
+
+KuiCollapseButton.propTypes = {
+  className: PropTypes.string,
+  direction: PropTypes.oneOf(DIRECTIONS).isRequired
+};
```

Comment:
You are right, DIRECTIONS is part of the interface, even if it's not used directly by the user.
Ok.

Probably later if the UI framework is more stable this interface can be extended/replaced by named collapse buttons like `KuiCollapseButtonUp` (the same idea as I said about `KuiEventSymbol`). It's a shorter syntax but not so flexible than this.